boom/bootloader.py

```
@@ -2258,16 +2299,16 @@ def grub_users(self):
         return self._entry_data_property(BOOM_ENTRY_GRUB_USERS)
 
     @grub_users.setter
-    def grub_users(self, grub_users):
+    def grub_users(self, grub_users: str):
         bls_key = KEY_MAP[BOOM_ENTRY_GRUB_USERS]
-        if not self._have_optional_key(bls_key):
+        if self._osp and not self._have_optional_key(bls_key):
             raise ValueError(
                 f"OsProfile os_id={self._osp.disp_os_id} does not allow '{bls_key}'"
             )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)